packages/api/src/bsky-agent.ts

```diff
@@ -491,34 +567,94 @@ export class BskyAgent extends AtpAgent {
     return prefs
   }
 
+  async setSavedFeedsV2(savedFeeds: AppBskyActorDefs.SavedFeed[]) {
+    savedFeeds.forEach(validateSavedFeed)
+    const uniqueSavedFeeds = new Map<string, AppBskyActorDefs.SavedFeed>()
+    savedFeeds.forEach((feed) => uniqueSavedFeeds.set(feed.id, feed))
+    return updateSavedFeedsV2Preferences(this, () =>
```

Comment:
it seems like this would re-order the feeds, right?

Reply:
Ah good call, if there are duplicates it should overwrite the first instance instead of appending. I think it should probably append a new entry if a duplicate `id` is found.

Or is there another case I'm missing?

Reply:
do `Map`s preserve order? I figured that the values would end up coming out in order of their key (but honestly i'm not sure how it works lol)

Reply:
Oh yep yep, good shout, they do indeed https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/values